<Type Name="StackOverflowException" FullName="System.StackOverflowException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="bb807895a73fd32279843a0dc0c00f2b0ed44976" />
    <Meta Name="ms.sourcegitcommit" Value="d877ae76e9e11799bf919379507239e2c4072742" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="zh-TW" />
    <Meta Name="ms.lasthandoff" Value="08/09/2018" />
    <Meta Name="ms.locfileid" Value="39809688" />
  </Metadata>
  <TypeSignature Language="C#" Value="public sealed class StackOverflowException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable sealed beforefieldinit StackOverflowException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.StackOverflowException" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class StackOverflowException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class StackOverflowException sealed : SystemException" />
  <TypeSignature Language="F#" Value="type StackOverflowException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="2194d-101">例外狀況，擲回於執行堆疊因為包含過多巢狀方法呼叫而溢位時。</span>
      <span class="sxs-lookup">
        <span data-stu-id="2194d-101">The exception that is thrown when the execution stack overflows because it contains too many nested method calls.</span>
      </span>
      <span data-ttu-id="2194d-102">這個類別無法被繼承。</span>
      <span class="sxs-lookup">
        <span data-stu-id="2194d-102">This class cannot be inherited.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="2194d-103"><xref:System.StackOverflowException> 執行堆疊溢位錯誤，通常會發生的非常深或無限遞迴函式會擲回。</span><span class="sxs-lookup"><span data-stu-id="2194d-103"><xref:System.StackOverflowException> is thrown for execution stack overflow errors, typically in case of a very deep or unbounded recursion.</span></span>  
  
 <span data-ttu-id="2194d-104"><xref:System.StackOverflowException> 會使用 HRESULT COR_E_STACKOVERFLOW，其值 0x800703E9。</span><span class="sxs-lookup"><span data-stu-id="2194d-104"><xref:System.StackOverflowException> uses the HRESULT COR_E_STACKOVERFLOW, which has the value 0x800703E9.</span></span> <span data-ttu-id="2194d-105"><xref:System.Reflection.Emit.OpCodes.Localloc> Intermediate language (IL) 指令則會擲回<xref:System.StackOverflowException>。</span><span class="sxs-lookup"><span data-stu-id="2194d-105">The <xref:System.Reflection.Emit.OpCodes.Localloc> intermediate language (IL) instruction throws <xref:System.StackOverflowException>.</span></span> <span data-ttu-id="2194d-106">如需初始屬性值的清單<xref:System.StackOverflowException>資訊，請參閱<xref:System.StackOverflowException.%23ctor%2A>建構函式。</span><span class="sxs-lookup"><span data-stu-id="2194d-106">For a list of initial property values for a <xref:System.StackOverflowException> object, see the <xref:System.StackOverflowException.%23ctor%2A> constructors.</span></span>  
  
## <a name="version-considerations"></a><span data-ttu-id="2194d-107">版本的考量</span><span class="sxs-lookup"><span data-stu-id="2194d-107">Version Considerations</span></span>  
 <span data-ttu-id="2194d-108">在.NET Framework 1.0 和 1.1 中，您可以攔截<xref:System.StackOverflowException>物件 （例如，從無限制的遞迴復原）。</span><span class="sxs-lookup"><span data-stu-id="2194d-108">In the .NET Framework 1.0 and 1.1, you could catch a <xref:System.StackOverflowException> object (for example, to recover from unbounded recursion).</span></span> <span data-ttu-id="2194d-109">從.NET Framework 2.0 開始，您不能攔截<xref:System.StackOverflowException>物件`try` / `catch`區塊，以及對應的處理序終止的預設值。</span><span class="sxs-lookup"><span data-stu-id="2194d-109">Starting with the .NET Framework 2.0, you can’t catch a <xref:System.StackOverflowException> object with a `try`/`catch` block, and the corresponding process is terminated by default.</span></span> <span data-ttu-id="2194d-110">因此，您應該撰寫您的程式碼來偵測並防止堆疊溢位。</span><span class="sxs-lookup"><span data-stu-id="2194d-110">Consequently, you should write your code to detect and prevent a stack overflow.</span></span> <span data-ttu-id="2194d-111">例如，如果您的應用程式相依於遞迴，使用計數器或狀態的條件來終止遞迴迴圈。</span><span class="sxs-lookup"><span data-stu-id="2194d-111">For example, if your app depends on recursion, use a counter or a state condition to terminate the recursive loop.</span></span> <span data-ttu-id="2194d-112">下列範例會使用計數器，以確保遞迴數目呼叫`Execute`方法不會超過 MAX_RECURSIVE_CALLS 常數所定義的最大值。</span><span class="sxs-lookup"><span data-stu-id="2194d-112">The following example uses a counter to ensure that the number of recursive calls to the `Execute` method do not exceed a maximum defined by the MAX_RECURSIVE_CALLS constant.</span></span>  
  
 [!code-csharp[System.StackOverflowException.Class#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.stackoverflowexception.class/cs/example1a.cs#1)]
 [!code-vb[System.StackOverflowException.Class#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.stackoverflowexception.class/vb/example1a.vb#1)]  
  
> [!NOTE]
>  <span data-ttu-id="2194d-113">套用<xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute>屬性的方法，就會擲回<xref:System.StackOverflowException>沒有任何作用。</span><span class="sxs-lookup"><span data-stu-id="2194d-113">Applying the <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> attribute to a method that throws a <xref:System.StackOverflowException> has no effect.</span></span> <span data-ttu-id="2194d-114">您仍然無法處理的例外狀況，從使用者程式碼。</span><span class="sxs-lookup"><span data-stu-id="2194d-114">You still cannot handle the exception from user code.</span></span>  
  
 <span data-ttu-id="2194d-115">如果您的應用程式裝載 common language runtime (CLR)，它可以指定 CLR 應該卸載的應用程式定義域，其中就會發生堆疊溢位例外狀況，而讓對應的程序繼續執行。</span><span class="sxs-lookup"><span data-stu-id="2194d-115">If your app hosts the common language runtime (CLR), it can specify that the CLR should unload the application domain where the stack overflow exception occurs and let the corresponding process continue.</span></span> <span data-ttu-id="2194d-116">如需詳細資訊，請參閱 < [ICLRPolicyManager 介面](~/docs/framework/unmanaged-api/hosting/iclrpolicymanager-interface.md)。</span><span class="sxs-lookup"><span data-stu-id="2194d-116">For more information, see [ICLRPolicyManager Interface](~/docs/framework/unmanaged-api/hosting/iclrpolicymanager-interface.md).</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="2194d-117">初始化 <see cref="T:System.StackOverflowException" /> 類別的新執行個體。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-117">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="2194d-118">初始化 <see cref="T:System.StackOverflowException" /> 類別的新執行個體，設定新執行個體的 <see cref="P:System.Exception.Message" /> 屬性為描述錯誤的系統提供訊息，例如「要求的作業造成堆疊溢位」。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-118">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class, setting the <see cref="P:System.Exception.Message" /> property of the new instance to a system-supplied message that describes the error, such as "The requested operation caused a stack overflow."</span>
          </span>
          <span data-ttu-id="2194d-119">此訊息會考量目前的系統文化特性。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-119">This message takes into account the current system culture.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="2194d-120">下表顯示 <xref:System.StackOverflowException> 執行個體的初始屬性值。</span><span class="sxs-lookup"><span data-stu-id="2194d-120">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="2194d-121">屬性</span><span class="sxs-lookup"><span data-stu-id="2194d-121">Property</span></span>|<span data-ttu-id="2194d-122">值</span><span class="sxs-lookup"><span data-stu-id="2194d-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="2194d-123">`null`。</span><span class="sxs-lookup"><span data-stu-id="2194d-123">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="2194d-124">當地語系化的錯誤訊息字串。</span><span class="sxs-lookup"><span data-stu-id="2194d-124">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new StackOverflowException : string -&gt; StackOverflowException" Usage="new System.StackOverflowException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="2194d-125">描述錯誤的 <see cref="T:System.String" />。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-125">A <see cref="T:System.String" /> that describes the error.</span>
          </span>
          <span data-ttu-id="2194d-126">訊息的內容必須能讓人瞭解。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-126">The content of message is intended to be understood by humans.</span>
          </span>
          <span data-ttu-id="2194d-127">這個建構函式的呼叫端必須確保這個字串已經被當地語系化 (為了目前系統的文化特性)。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-127">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="2194d-128">使用指定的錯誤訊息，初始化 <see cref="T:System.StackOverflowException" /> 類別的新執行個體。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-128">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class with a specified error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="2194d-129">下表顯示 <xref:System.StackOverflowException> 執行個體的初始屬性值。</span><span class="sxs-lookup"><span data-stu-id="2194d-129">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="2194d-130">屬性</span><span class="sxs-lookup"><span data-stu-id="2194d-130">Property</span></span>|<span data-ttu-id="2194d-131">值</span><span class="sxs-lookup"><span data-stu-id="2194d-131">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="2194d-132">null 參考 (在 Visual Basic 中為 `Nothing`)。</span><span class="sxs-lookup"><span data-stu-id="2194d-132">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="2194d-133">錯誤訊息字串。</span><span class="sxs-lookup"><span data-stu-id="2194d-133">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new StackOverflowException : string * Exception -&gt; StackOverflowException" Usage="new System.StackOverflowException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="2194d-134">解釋例外狀況原因的錯誤訊息。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-134">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="2194d-135">做為目前例外狀況發生原因的例外狀況。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-135">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="2194d-136">如果 <c>innerException</c> 參數不是 Null 參考 (在 Visual Basic 中為 <see langword="Nothing" />)，則會在處理內部例外狀況的 <see langword="catch" /> 區塊中引發目前的例外狀況。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-136">If the <c>innerException</c> parameter is not a null reference (<see langword="Nothing" /> in Visual Basic), the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="2194d-137">使用指定的錯誤訊息以及造成此例外狀況的內部例外狀況的參考，初始化 <see cref="T:System.StackOverflowException" /> 類別的新執行個體。</span>
          <span class="sxs-lookup">
            <span data-stu-id="2194d-137">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="2194d-138">直接因為前一個例外狀況而擲回的例外狀況可能包含對 <xref:System.Exception.InnerException%2A> 屬性中前一個例外狀況的參考。</span><span class="sxs-lookup"><span data-stu-id="2194d-138">An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="2194d-139"><xref:System.Exception.InnerException%2A> 屬性會傳回與傳入建構函式中相同的值；如果 `Nothing` 屬性不提供內部例外值給建構函式，則傳回 null 參考 (在 Visual Basic 中為 <xref:System.Exception.InnerException%2A>)。</span><span class="sxs-lookup"><span data-stu-id="2194d-139">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="2194d-140">下表顯示 <xref:System.StackOverflowException> 執行個體的初始屬性值。</span><span class="sxs-lookup"><span data-stu-id="2194d-140">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="2194d-141">屬性</span><span class="sxs-lookup"><span data-stu-id="2194d-141">Property</span></span>|<span data-ttu-id="2194d-142">值</span><span class="sxs-lookup"><span data-stu-id="2194d-142">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="2194d-143">內部例外狀況參考。</span><span class="sxs-lookup"><span data-stu-id="2194d-143">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="2194d-144">錯誤訊息字串。</span><span class="sxs-lookup"><span data-stu-id="2194d-144">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
  </Members>
</Type>